linux/datadog_wrapper

```diff
@@ -1,96 +1,165 @@
+        export CORECLR_PROFILER="{846F5F1C-F9AE-4B07-969E-05C26BC060D8}"
+        export CORECLR_PROFILER_PATH="${DD_DIR}/Datadog.Trace.ClrProfiler.Native.so"
+        export DD_DOTNET_TRACER_HOME="${DD_DIR}"
+    fi
+}
 
-setUpCommonEnv() {
-    echo "Creating Datadog directory"
-    mkdir -p $DD_DIR && cd $DD_DIR || return
+setEnvVars() {
+    if [ -z "$DD_DIR" ]; then
+        DD_DIR="/home/datadog"
+    fi
 
-    FILE=trace-agent-${DD_AAS_LINUX_VERSION}.tar.gz
+    if [ -z "$DD_BINARIES_URL" ]; then
+        DD_BINARIES_URL="https://github.com/DataDog/datadog-aas-extension/releases/download"
+    fi
 
-    if [ ! -f $FILE ]; then 
+    if [ -z "$DD_AAS_LINUX_VERSION" ]; then
+        DD_AAS_LINUX_VERSION="linux-v0.1.4-beta"
+    fi
 
-        if [ -f "trace-agent" ]; then 
-            echo "Removing old versions of the trace agent"
-            rm trace-agent* 
-        fi
+    if [ -z "$DD_TRACE_ENABLED" ]; then
+        DD_TRACE_ENABLED="true"
+    fi
 
-        echo "Downloading required Datadog binaries from $DD_BINARIES_URL/$DD_AAS_LINUX_VERSION/$FILE"
-        if curl -L --fail $DD_BINARIES_URL/$DD_AAS_LINUX_VERSION/$FILE -o $FILE; then echo
-            echo "Decompressing files"
-            tar -zxf $FILE || return
-        else 
-            echo "Failed to download the Datadog binary succesfully."
-            return
-        fi
-        
-    else
-        echo "Version $DD_AAS_LINUX_VERSION of the trace agent previously installed"
+    if [ -z "$DD_CUSTOM_METRICS_ENABLED" ]; then
+        DD_CUSTOM_METRICS_ENABLED="false"
     fi
-    
-    echo "Starting trace agent"
-    ./trace-agent &
-    TRACER_PID=$!
-    trap 'kill $TRACER_PID ; exit' INT
 }
 
-setUpNodeEnv() {
-    echo "Setting up Datadog tracing for Node"
+getBinaries() {
+    if [ "$DD_TRACE_ENABLED" ]; then
+        echo "Setting up the trace agent"
+        setUpBinary "trace-agent" & STARTUP_PIDS+=($!)
+    fi
 
-    setUpCommonEnv || return
+    if [ "$DD_CUSTOM_METRICS_ENABLED" ]; then
+        echo "Setting up DogStatsD"
+        setUpBinary "dogstatsd" & STARTUP_PIDS+=($!)
+    fi
+}
 
-    echo "Installing Node tracer"
-    yarn add dd-trace || return
+setUpBinary() {
+    BINARY="$1"
+    ARCHIVED_BINARY="$BINARY-${DD_AAS_LINUX_VERSION}.tar.gz"
 
-    ORIG_NODE_OPTIONS=$NODE_OPTIONS
-    export NODE_OPTIONS="--require=$DD_DIR/node_modules/dd-trace/init $ORIG_NODE_OPTIONS"
+    check_file_exists_or_curl "$ARCHIVED_BINARY" "$BINARY"
+    # do not start the binary if curling failed
+    if [ "$?" -ne 0 ]; then
+        echo "Curling $BINARY was unsuccesful and the binary will not be initialized"
+        return
+    fi
 
-    # confirm updates to NODE_OPTIONS
-    node --help >/dev/null || (export NODE_OPTIONS=$ORIG_NODE_OPTIONS && return)
+    echo "Starting $BINARY"
+    if [ $BINARY = "dogstatsd" ]; then
+        ./$BINARY start &
+        trap 'kill $! : exit' INT
+        return
+    fi
+
+    ./"$1" &
+    trap 'kill $! : exit' INT
```

Comment:
I feel like there's gotta be a better way to be able to reuse/share more of this code. For instance, what if you add a `$2` for any extra start command args? So you'd do `setUpBinary trace-agent` or `setUpBinary dogstatsd start`. And then the actual starting would change from `./$BINARY &` to `./$BINARY $EXTRA_START_ARGS &`. Something like that?